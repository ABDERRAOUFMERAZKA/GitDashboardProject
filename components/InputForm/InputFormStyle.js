import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
      width: 280,
      borderColor: '#fff'
  }
});