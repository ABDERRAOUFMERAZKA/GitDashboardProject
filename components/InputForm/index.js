import React, {Fragment} from 'react';
import { Button, TextInput } from 'react-native'
import PropTypes from 'prop-types'
import styles from './InputFormStyle'

// All Strings are saved in constant.js
import {INPUT_FORM} from '../../constant'


const InputForm = ({
    onChangeText,
    validateName
}) => {
  return (
    <Fragment>
      <TextInput
        placeholder={INPUT_FORM.input_text.placeholder}
        borderBottomColor={INPUT_FORM.input_text.border_color}
        style={styles.container}
        onChangeText={onChangeText}
      />
       <Button
         title={INPUT_FORM.button.title}
         color={INPUT_FORM.button.color}
         onPress={validateName}
      />
    </Fragment>
  );
};

InputForm.propTypes = {
  onChangeText: PropTypes.func.isRequired,
  validateName: PropTypes.func.isRequired,
}

export default InputForm;
