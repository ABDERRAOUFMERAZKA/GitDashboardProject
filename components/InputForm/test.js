import React from 'react';
import InputForm from './index';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<InputForm />).toJSON();
  expect(tree).toMatchSnapshot();
})