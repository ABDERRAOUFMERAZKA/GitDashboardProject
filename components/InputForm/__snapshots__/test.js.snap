// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <TextInput
    allowFontScaling={true}
    borderBottomColor="fff"
    placeholder="Please enter github user"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "#fff",
        "width": 280,
      }
    }
    underlineColorAndroid="transparent"
  />,
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    clickable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#f194ff",
            },
          ]
        }
      >
        Search
      </Text>
    </View>
  </View>,
]
`;
